### PR TITLE
A0-223: launching chain in CI

### PR DESCRIPTION
**Pull request by fbielejec**
This PR is a skeleton for future more evolved e2e/integration tests.

**Brief description:**

- on every PR opened:
  - CI runs a check, format + lint and unit test suite

- on every PR opened that targets the main branch
  - check, format + lint and unit test
  - integration test suite
   
- on push / merge to main
  - CI deploys new node devnet versions 
  - conditional on a manual approval from one of the team members CI deploys new testnet version

### CODE REVIEW

File: docker/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+services:
+
+  damian:
+    image: aleph-node:latest
+    container_name: damian
+    entrypoint: ["/bin/sh","-c"]
+    command:
+      - |
+         ./aleph-node purge-chain --chain testnet1 --base-path /tmp/Damian -y
+         ./aleph-node --validator --chain testnet1 --base-path /tmp/Damian --name Damian --node-key-file /tmp/Damian/libp2p_secret --rpc-port 9933 --ws-port 9944 --port 30334 --rpc-cors all --rpc-methods Safe --execution Native --no-prometheus --no-telemetry --reserved-only --reserved-nodes /ip4/172.28.1.1/tcp/30334/p2p/12D3KooWAbWDCsZiA6EXrXh33E7hBYUFK53b5CSajoPsYL5n2Zdm /ip4/172.28.1.2/tcp/30334/p2p/12D3KooWQdQW1Laeq869b4pdDGSVZebbjY9h9wmdNnauFeFWUKup /ip4/172.28.1.3/tcp/30334/p2p/12D3KooWD4UoT2euDnFwnSRhfUjMt6DWkMTGWV8hQm2tuNgX4j3V /ip4/172.28.1.4/tcp/30334/p2p/12D3KooWQzGBGxYx3Su9EaFMS1qtVHc2zSYdTmgRTeRGjGXj9uNg --session-period 25
```
Comment (mike1729) [inline docker/docker-compose.yml:11]: I propose to set miliseconds_per_block to 1000.

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -85,9 +85,62 @@ jobs:
...
+      - name: Run consensus party
+        run: |
+          docker-compose -f docker/docker-compose.yml up &
+
+      - name: Verify nodes respond
+        run: |
+          validators=( damian tomasz zbyszko hansu )
+
+          for validator in "${validators[@]}"; do
+            docker run --network container:$validator appropriate/curl:latest --retry 3 --retry-delay 10 -H "Content-Type: application/json" -d '{"id":1, "jsonrpc":"2.0", "method": "rpc_methods"}' http://127.0.0.1:9933
```
Comment (mike1729) [inline .github/workflows/deploy.yml:124]: Could you break this line with '\'? Also, a small comment what this call exactly do would be helpful.

**Review by mike1729 — CHANGES_REQUESTED** [review]
Two tiny requests.

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (fbielejec) [reply .github/workflows/deploy.yml]: Done

File: docker/docker-compose.yml (hunk 1 above)
Comment (fbielejec) [reply docker/docker-compose.yml]: Done

**Review by mike1729 — APPROVED** [review]

**Review by maciejnems — APPROVED** [review]

**Pull request merged**